This time it's the other way round: both GNU libc and musl libc
have <endian.h> so use defined(__linux__) to use this header
file for setting the target's endianness.

--- a/src/osgPlugins/osc/osc/OscHostEndianness.h	2015-07-23 13:14:00.000000000 +0200
+++ b/src/osgPlugins/osc/osc/OscHostEndianness.h	2015-10-31 06:39:02.503234496 +0100
@@ -51,7 +51,7 @@
 
 #else
 
-    #if defined(__GLIBC__) || defined(__ANDROID__) || defined(__CYGWIN__)
+    #if defined(__linux__) || defined(__ANDROID__) || defined(__CYGWIN__)
         #include <endian.h>
         #if (__BYTE_ORDER == __LITTLE_ENDIAN)
             #ifndef __LITTLE_ENDIAN__
